clients/drcachesim/tools/schedule_stats.cpp

```diff
@@ -361,8 +361,10 @@ schedule_stats_t::parallel_shard_memref(void *shard_data, const memref_t &memref
     // We use <workload,tid> to detect switches (instead of input_id) to handle
     // core-sharded-on-disk.  However, we still prefer the input_id ordinal
     // for the letters.
-    int64_t workload_id = shard->stream->get_workload_id();
     int64_t tid = shard->stream->get_tid();
+    int64_t workload_id = TESTANY(OFFLINE_FILE_TYPE_CORE_SHARDED, shard->filetype)
```

Comment:
Worth updating the documentation of OFFLINE_FILE_TYPE_CORE_SHARDED to note these subtle limitations.

Maybe also at memtrace_stream_t::get_workload_id. Or would get_workload_id return -1 for csod (" If not implemented for the current mode, -1 is returned")

Reply:
Done and done (no it won't return -1 it will return 0 as it sees a single workload).